src/entity.ts

```diff
@@ -52,27 +52,44 @@ export namespace entity {
    * provided.
    *
    * @class
-   * @param {number} value The double value.
+   * @param {number|string} value The double value.
    *
    * @example
    * const {Datastore} = require('@google-cloud/datastore');
    * const datastore = new Datastore();
    * const aDouble = datastore.double(7.3);
    */
-  export class Double {
+  export class Double extends Number {
+    private _entityPropertyName: string | undefined;
     type: string;
     value: number;
-    constructor(value: number) {
+    constructor(value: number | string | ValueProto) {
+      super(typeof value === 'object' ? value.doubleValue : value);
+
       /**
        * @name Double#type
        * @type {string}
        */
       this.type = 'DatastoreDouble';
+
+      this._entityPropertyName =
```

Comment:
Do we really need to keep the property name ?
If yes it would be good to comment why.

Reply:
Honestly, I am not sure what this is for, but I mirrored behavior in the Int implementation. Maybe @stephenplusplus recalls?

Reply:
I haven't checked the use cases but it seems dangerous

if an entity has the form `{oldValue: Int, newValue: Int}` and you do (with wrapped Number):

```typescript
const [entity] = await ds.get(...);

entity.oldValue = entity.newValue;
entity.newValue = ds.Int(123);

await ds.save(entity);
```

I can only imagine that keeping the entity property name would cause problems as `oldValue` would have a reference to `"newValue"`.

I am not sure if it is actually a problem in the current code but at least it makes it easy to shoot yourself in the foot.